tests/load_straw_prep_data.py

```diff
@@ -0,0 +1,340 @@
+
+    cpal_list, straw_information, cpal_prefix_list = get_straws(problem_files)
+
+    return cpal_list, straw_information, cpal_prefix_list
+
+    
+    
+def update_straw_table(straw_information, cpal_prefix_list):
+    database = GetLocalDatabasePath()
+    engine = sqla.create_engine("sqlite:///" + database)
+    connection = engine.connect()
+    
+    # for all straws not present in straw table, add them
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        for y in range(len(straw_information[cpal])):
+            straw_id = straw_information[cpal][y]['id'][2::].lstrip('0')
+            batch = straw_information[cpal][y]['batch']
+            timestamp = int(straw_information[cpal][y]['time'])
+            
+            try:
+                straw = Straw.Straw(straw_id, batch, timestamp)
+            except:
+                print('Error saving straw ' + str(straw_id) + ' on cpal ' + str(cpal))
+
+    print('done updating straw table')
+    
+def update_straw_location_table(cpal_prefix_list):
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        cpalid = i['cpalid']
+        time = i['time']
+    
+        try:
+            straw_location = StrawLocation.query().filter(StrawLocation.location_type == 'CPAL').filter(StrawLocation.number == cpal).all()
+            if len(straw_location) == 0:
+                location = StrawLocation('CPAL',cpal,cpalid,None,False,time)
+                #location = Pallet('CPAL',cpal,cpalid,None,time)
+        except:
+            print('Error saving cpal ' + str(cpal))
+    
+    print('done updating straw location table')
+    
+def update_measurement_prep_table(cpal_prefix_list, straw_information):
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        cpalid = i['cpalid']
+        time = i['time']
+        
+        try:
+            for y in range(len(straw_information[cpal])):
+                straw_id = straw_information[cpal][y]['id'][2::].lstrip('0')
+                batch = straw_information[cpal][y]['batch']
+                paper_pull = straw_information[cpal][y]['grade']
+                timestamp = int(straw_information[cpal][y]['time'])
+            
+                
+                procedure = Procedure.StrawProcedure(2,cpalid,cpal,timestamp)
+                
+                prep_measurement = Prep.StrawPrepMeasurement(procedure, straw_id, paper_pull[-1], 1, timestamp)
+                prep_measurement.commit()
+        except:
+            print('Error saving to measurement prep table for cpal ' + str(cpal))
+
+            
+def update_straw_present_table(cpal_prefix_list, straw_information):
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        cpalid = i['cpalid']
+        time = i['time']
+        
+        straw_location = StrawLocation.query().filter(StrawLocation.location_type == 'CPAL').filter(StrawLocation.number == cpal).all()[0]
```

Comment:
one_or_none --> try-except / check on the None

Reply:
Modified accordingly. I checked, it returns none if none exists. It throws an error if multiple exist.